src/mlpack/methods/kmeans/hamerly_kmeans_impl.hpp

```diff
@@ -74,9 +79,13 @@ double HamerlyKMeans<DistanceType, MatType>::Iterate(const arma::mat& centroids,
     // First bound test.
     if (upperBounds(i) <= m)
     {
+      #pragma omp atomic
```

Comment:
Does this need to be atomic if we already have the reduction for `hamerlyPruned`?